### ISSUE TITLE
"Masking the solar disk" gallery example does not have a black mask

### ISSUE THREAD

**Issue by wtbarnes** [sunpy/sunpy#3007]
### Description
In the "Masking the solar disk" example, the mask is incorrectly colored. The mask is explicitly set to be black
```python
palette = aia.plot_settings['cmap']
palette.set_bad('black')
```
but it appears white in the generated gallery example. I suspect this is due to a change in the way matplotlib treats NaNs in imshow.

### Steps to Reproduce
See example gallery in docs.

**Comment by ankursikarwar** [comment]
Is it necessary to have the mask in black color? I think matplotlib sets the default color for NaNs as white. Because palette.set_bad() function is not having any effect on the color of the mask.

**Comment by wtbarnes** [comment]
Yeah I guess the issue I'm pointing out is that, as you say, `palette.set_bad('black')` has no effect on the color of the mask.

**Comment by ankursikarwar** [comment]
I found that the plot function in map/mapbase.py accepts `**imshow_kwargs`  but doesn't then pass them to the imshow function. Will make the respective changes and submit a pull request.

**Comment by ankursikarwar** [comment]
I found that the plot function in map/mapbase.py accepts **imshow_kwargs but doesn't then pass them to the imshow function. Shall I make the respective changes and submit a pull request?

**Comment by ankursikarwar** [comment]
I think this is caused because of normalization because when I set `norm=None` then `palette.set_bad('black')` works fine.

Before setting `norm=None` : 
![54865925-6307c500-4d65-11e9-8f73-f116e136e03b](https://user-images.githubusercontent.com/32908468/54869849-77c77580-4dc4-11e9-9430-f1311c6d7704.png)


After setting `norm=None` : 
![54865924-55523f80-4d65-11e9-89c1-06cab13cb878](https://user-images.githubusercontent.com/32908468/54869858-8dd53600-4dc4-11e9-9474-3e472ca5cd2b.png)
But setting `norm=None` causes the edges of the image to become dark.

**Comment by nabobalis** [comment]
Yeah, so we need to figure out a way to get them both to work. I assume this is a bug within our plotting code.

**Comment by manurare** [comment]
Hi. I found out that the mask values are having white color because of the default norm indicated when reading AIA_171_IMAGE. This can be observed in the _mpl_normalize.py_ in the Astropy library.  At the line 97 they indicate that the values inside the mask should be filled with the largest value of the data array (which will be white):
`            values = values.filled(self.vmax)
`
and it should be with the lowest one:
`            values = values.filled(self.vmin)
`

then you will get the right results

![Black mask](https://user-images.githubusercontent.com/43776720/56436021-bcb6cd00-62da-11e9-9a73-d26d8c9d5520.png)

**Comment by nabobalis** [comment]
Would it be possible for you to open a PR with this fix?

Ideally with a comment as to why and a link to the astropy example/docs that explain it?

**Comment by manurare** [comment]
But how do I open a PR if the bug is located in the AstroPy library? Can I do it inside sunpy?

**Comment by nabobalis** [comment]
So to get that image you had to change astropy code?

**Comment by manurare** [comment]
Yes, inside astropy/visualization in the _mpl_normalize.py_ I changed the line shown above. Astropy considers the pixels inside the mask by default as white while in the example it is asked for black.

**Comment by nabobalis** [comment]
You could open an issue on astropy asking if that is intended for masks. Otherwise, we need to change/update the normalization code inside sunpy.

**Comment by manurare** [comment]
I contacted them and I think it is intended because most of the space images have black background and therefore as a high contrast they consider the mask white. Nevertheless they might be considering on the possibility of customizing the color of the maks. Here is the opened [issue](https://github.com/astropy/astropy/issues/8603). I dont know if this really answers SunPy problem though

**Comment by nabobalis** [comment]
Well in that case it seems we would need to see if we can change that inside SunPy without touching astropy. Unless we can come up with a patch to astropy to create a user setting we can change.

**Comment by ravindra1307** [comment]
To get desired output without touching astropy. my solution is to change norm values as that gives approximately same effect as desired.
 this is the result after changing norm value.

![sunpy3](https://user-images.githubusercontent.com/21353494/56462047-f4586e80-63d9-11e9-9e79-98a3b9db6cd2.png)

**Comment by nabobalis** [comment]
That was the change in the PR I closed? If so I apologize, you should reopen it and we can get it merged.

**Comment by ravindra1307** [comment]
thanks, i will reopen it.

**Comment by ayshih** [comment]
The above discussion was actually all incorrect.  The bug was that our default normalizer had its clipping turned on due to a change in Astropy's defaults.  This has now been fixed via #3427, and the mask now has the intended color in [our gallery example](https://docs.sunpy.org/en/latest/generated/gallery/computer_vision_techniques/mask_disk.html).  Closing.
